core/src/main/java/io/confluent/rest/Application.java

```diff
@@ -245,6 +244,14 @@ protected void doStop() throws Exception {
       context.setBaseResource(staticResources);
     }
 
+    if(config.getBoolean(RestConfig.ENABLE_GZIP_COMPRESSION_CONFIG)) {
+      FilterHolder gzipFilter = new FilterHolder(AsyncGzipFilter.class);
+      // do not check if .gz file already exists for the requested resource
+      gzipFilter.setInitParameter("checkGzExists", "false");
+      gzipFilter.setInitParameter("methods", "GET,POST");
+      context.addFilter(gzipFilter, "/*", null);
```

Comment:
I'm not sure if that level of granularity is necessary, do you have a particular use-case in mind that would warrant not turning it on? The only reason I can think of is if we served incompressible binary data from dynamic resources.